topic-auto-create/README.md

```diff
@@ -0,0 +1,134 @@
+
+## Prerequisites for this example
+
+- Installed Docker and `docker-compose`
+- Installed `curl` (and optionally `json_pp`)
+- Cloned or downloaded this example
+
+## Usage
+
+You can find the documentation of the Change Data Topics Auto-creation in the Debezium Docs[] 
+
+### Start Docker Containers
+
+To spin up some example infrastructure like Kafka, Debezium and a Postgres database you
+have to execute from this example directory (`./topic-auto-create` directory):
+
+```bash
+$ DEBEZIUM_VERSION=1.3 docker-compose up -d
+
+  Creating network "topic-auto-create_default" with the default driver
+  Creating topic-auto-create_zookeeper_1 ... done
+  Creating topic-auto-create_postgres_1  ... done
+  Creating topic-auto-create_kafka_1     ... done
+  Creating topic-auto-create_connect_1   ... done
+``` 
+
+### Connector Config
+
+Then we use the following connector config for a Postgres connector to capture the `inventory`
+schema where the default config for topics should be `replication.factor = 1`, `partitions = 5`,
+the topic should be key compacted with `cleanup.policy = "compact"`, and all messages should
+be forced to GZIP compression on harddisk with `compression.type = "gzip"`.
+
+Some topics of captured tables that have a name starting with `product` should go to topics
+that have a retention time of 3 months / 90 days with `cleanup.policy": "delete"` and
+`retention.ms = 7776000000`, `replication.factor = 1`, `partitions = 20`, and just use the
+compression format that's chosen by the producer with `compression.type": "producer"`.
+
+```json
+{
+    "name": "inventory-connector",
+    "config": {
+        "connector.class": "io.debezium.connector.postgresql.PostgresConnector",
+        "tasks.max": 1,
+        "database.hostname": "postgres",
+        "database.port": 5432,
+        "database.user": "postgres",
+        "database.password": "postgres",
+        "database.dbname" : "postgres",
+        "database.server.name": "dbserver1",
+        "schema.include.list": "inventory",
+
+        "topic.creation.default.replication.factor": 1,
+        "topic.creation.default.partitions": 5,
+        "topic.creation.default.cleanup.policy": "compact",
+        "topic.creation.default.compression.type": "gzip",
+        "topic.creation.default.delete.retention.ms" : 2592000000,
+
+        "topic.creation.groups": "productlog",
+
+        "topic.creation.productlog.include": "dbserver1\\.inventory\\.product.*",
+        "topic.creation.productlog.replication.factor": 1,
+        "topic.creation.productlog.partitions": 10,
+        "topic.creation.productlog.cleanup.policy": "delete",
+        "topic.creation.productlog.compression.type": "producer",
+        "topic.creation.productlog.retention.ms": 7776000000
+    }
+}
+```
+
+### Sending connector config
+
+```bash
+$ curl -X POST -H "Accept: application/json" -H "Content-Type: application/json" -d @connector.json "http://localhost:8083/connectors" | json_pp
```

Comment:
Let's use PUT, as it's idempotent and generally preferred in our examples (I know it's not done yet consistently).

Reply:
I never use PUT. With POST and having the connector name inside the JSON file is much more reliable and idempotent compared to the PUT way from my experiences when you have your connector config checked-in to VCS.

Reply:
Also for the examples I think POST-ing the full config with connector name in JSON makes it "Luser-proof".

Reply:
Done. But let me challenge you on that topic later!